dummy2-Now with couplers! (#508)
qiboteam/qibolab: doc/source/tutorials/lab.rst

        )
    )

Some architectures may also have coupler qubits that mediate the interactions.
We can also interact with them defining the :class:`qibolab.couplers.Coupler` objects.

> Then we add them to their corresping :class:`qibolab.qubits.QubitPair` objects according
> to the chip topology. Again we neglected or characterization parameters associated to the

Reviewer: ```suggestion
Then we add them to their corresponding :class:`qibolab.qubits.QubitPair` objects according
to the chip topology. We neglected characterization parameters associated to the
```

                threshold: 0.0002694329123116206
                iq_angle: 4.912447775569025

And in the case of having a chip with coupler qubits

> we need to following changes to the previous runcard:

Reviewer: ```suggestion
we need the following changes to the previous runcard:
```

.. code-block::  yaml

    qubits: [0, 1]
    couplers: [0]

>     topology: {0: [0, 1]}

Reviewer: Maybe we can add a note here that this maps coupler to qubit pair. I am not sure if it will be clear to everyone.

    native_gates:
        two_qubit:
            0-1:
                CZ:
                - duration: 30
                amplitude: 0.6025
                shape: Rectangular()

>                 qubit: 2

Reviewer: ```suggestion
                qubit: 0
```
we should change all qubits to 0, 1 because this platform is defined with two qubits only.

>                 - duration: 30
>                 amplitude: 0.6025
>                 shape: Rectangular()
>                 qubit: 2
>                 relative_start: 0
>                 type: qf

Reviewer: I am not sure but I think the indentation needs to be
```suggestion
                - duration: 30
                  amplitude: 0.6025
                  shape: Rectangular()
                  qubit: 2
                  relative_start: 0
                  type: qf
```
At least I got some issues when I try to use it as it is.

                - type: virtual_z
                phase: -1
                qubit: 1
                - type: virtual_z
                phase: -3
                qubit: 2

                - type: coupler

>                 phase: -3

Reviewer: ```suggestion
```
I think here there is no phase for these pulses. I got an error when I try to create the platform. If we need a phase maybe something is missing from the code.

                  duration: 40
                  amplitude: 0.1
                  shape: Rectangular()

>                   coupler: 1

Reviewer: ```suggestion
                  coupler: 0
```

            "my_platform", qubits, pairs, instruments, settings, resonator_type="2D"
        )

With the following additions for coupler architectures:

> .. code-block::  python
>     def create():

Reviewer: ```suggestion
.. code-block::  python

    def create():
```
needs an empty line here, otherwise sphinx complains when I compile the docs.

        channels |= Channel("ch1out", port=instrument["o1"])
        channels |= Channel("ch1in", port=instrument["i1"])
        channels |= Channel("ch2", port=instrument["o2"])
        channels |= Channel("ch3", port=instrument["o3"])
        channels |= Channel("chf1", port=instrument["o4"])
        channels |= Channel("chf2", port=instrument["o5"])
        channels |= Channel("chfc0", port=instrument["o6"])

        # create ``Qubit`` and ``QubitPair`` objects by loading the runcard
        runcard = load_runcard(Path(__file__).parent / "my_platform.yml")
        qubits, couplers, pairs = load_qubits(runcard)

        # assign channels to the qubit
        for q in range(2):
            qubits[q].readout = channels["ch1out"]
            qubits[q].feedback = channels["ch1in"]
            qubits[q].drive = channels[f"ch{q + 2}"]
            qubits[q].flux = channels[f"chf{q + 1}"]

        # assign channels to the coupler
        couplers[0].flux = channels["chfc0"]

        # create dictionary of instruments
        instruments = {instrument.name: instrument}
        # load ``settings`` from the runcard
        settings = load_settings(runcard)
        return Platform(

>             "my_platform", qubits, pairs, instruments, settings, resonator_type="2D", couplers

Reviewer: ```suggestion
            "my_platform", qubits, pairs, instruments, settings, resonator_type="2D", couplers=couplers
```
otherwise Python will raise an error if you try to run this, because it is after the `resonator_type=`.
